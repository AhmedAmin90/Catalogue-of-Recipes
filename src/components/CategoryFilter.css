.CategoryFilter-select {
    color: #215330;
    letter-spacing: 2px;
  }

  .CategoryFilter-select > select {
      width: 200px;
      height: 40px;
      border-radius: 2px;
      cursor: pointer;
      font-size: 15px;
      letter-spacing: 2px;
  }
  
  .CategoryFilter-select > select > option {
      font-size: 15px;
      color: #215330;
      background-color: #DDDDDD;

  }