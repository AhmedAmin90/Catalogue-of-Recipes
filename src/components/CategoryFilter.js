import React from 'react';
import PropTypes from 'prop-types';
import { useSelector , useDispatch} from 'react-redux';
import * as actions from '../actions/index'
import './CategoryFilter.css'

function CategoryFilter({ handleFilter }) {
  const dispatch = useDispatch();
  const categories = useSelector(state => state.catReducer);
  const selectedCat = useSelector(state=> state.filterReducer);

  const handleChange = ({ target }) => {
    handleFilter(target.value);
  };


  return (
    <div className="CategoryFilter">
        <label htmlFor="fiter-category" className="CategoryFilter-select">
          Filter By Category
          <select onChange={handleChange} name="category" id="fiter-category">
            <option>{selectedCat}</option>
            {categories.map((cat) => <option key={cat} value={cat}>{cat}</option>)}
          </select>
        </label>
    </div>
  );
}

CategoryFilter.defaultProps = {
  handleFilter: () => {},
};

CategoryFilter.propTypes = {
  handleFilter: PropTypes.func,
};

export default CategoryFilter;