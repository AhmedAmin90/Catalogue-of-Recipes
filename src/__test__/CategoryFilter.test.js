import React ,  {useState} from 'react';
import { create } from 'react-test-renderer';
import { render , fireEvent , screen} from '@testing-library/react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { store } from '../index';
import {
    BrowserRouter as Router,
  } from 'react-router-dom';
import CategoryFilter from '../components/CategoryFilter';


test('renders correctly', () => {
  const tree = create(
    <Provider store={store}>
    <Router>
       <CategoryFilter handleFilter={() => {}} value="All"/>
    </Router>
  </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('rendered Main', () => {
    let renderedComponent;
    beforeEach(() => {
      renderedComponent = render(
        <Provider store={store}>
          <Router>
             <CategoryFilter handleFilter={() => {}} value="All" />
          </Router>
        </Provider>,
      );
    });

    it('has a div with class CategoryFilter', () => {
        const { container } = renderedComponent;
        const appElement = container.querySelector('.CategoryFilter');
        expect(appElement).toBeInTheDocument();
        screen.getByText(/All/);
      });

});

describe('Test handle Change' , ()=> {
  const CategoryFilter= ()=> {
    const [value, setValue] = useState('')
    
    return (
      <select  name="category" id="fiter-category">
          <option role="tab" value={value}>{value}</option>
      </select>
    )
  }
  const setup = () => {
    const component = render(<CategoryFilter  />)
    const input = component.getByRole('tab')
    return {
      input,
      ...component,
    }
  }
  test('Test changing value', () => {
    const {input} = setup()
    fireEvent.change(input, {target: {value: 'Test Value'}})
    expect(input.value).toBe('Test Value')
  })

  test('Choose from Array' , () => {
    const categories = ['All' , 'Beef' , 'Chicked']
    const {input} = setup()
    fireEvent.change(input, {target: {value: categories[0]}})
    expect(input.value).toBe('All');
    fireEvent.change(input, {target: {value: categories[2]}})
    expect(input.value).toBe('Chicked')
  })
  
})


