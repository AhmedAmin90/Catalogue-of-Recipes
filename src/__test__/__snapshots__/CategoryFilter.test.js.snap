// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="CategoryFilter"
>
  <label
    className="CategoryFilter-select"
    htmlFor="fiter-category"
  >
    Filter By Category
    <select
      id="fiter-category"
      name="category"
      onChange={[Function]}
    >
      <option>
        All
      </option>
    </select>
  </label>
</div>
`;
